controllers/status/status.go

```diff
@@ -210,7 +216,119 @@ func RemoveComponentCondition(conditions *[]conditionsv1.Condition, component st
 	conditionsv1.RemoveStatusCondition(conditions, conditionsv1.ConditionType(component+ReadySuffix))
 }
 
-// ModelRegistryStatus struct holds the status for the ModelRegistry component.
+// Condition represents the state of the operator's
+// reconciliation functionality.
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatus struct {
+	Name        string                  `json:"name,omitempty"`
+	DisplayName string                  `json:"displayname,omitempty"`
+	Version     version.OperatorVersion `json:"version,omitempty"`
+	RepoURL     string                  `json:"repourl,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentStatus struct {
+	Releases []ComponentReleaseStatus `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type CodeFlareStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DashboardStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type WorkbenchesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ModelMeshServingStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DataSciencePipelinesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KserveStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KueueStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type RayStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrustyAIStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
 type ModelRegistryStatus struct {
 	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
+	ComponentStatus     `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrainingOperatorStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentsStatus struct {
+	CodeFlare            *CodeFlareStatus            `json:"codeflare,omitempty"`
+	Dashboard            *DashboardStatus            `json:"dashboard,omitempty"`
+	DataSciencePipelines *DataSciencePipelinesStatus `json:"datasciencepipelines,omitempty"`
+	ModelMeshServing     *ModelMeshServingStatus     `json:"modelmeshserving,omitempty"`
+	ModelRegistry        *ModelRegistryStatus        `json:"modelregistry,omitempty"`
+	Kserve               *KserveStatus               `json:"kserve,omitempty"`
+	Kueue                *KueueStatus                `json:"kueue,omitempty"`
+	Ray                  *RayStatus                  `json:"ray,omitempty"`
+	TrustyAI             *TrustyAIStatus             `json:"trustyai,omitempty"`
+	TrainingOperator     *TrainingOperatorStatus     `json:"trainingoperator,omitempty"`
+	Workbenches          *WorkbenchesStatus          `json:"workbenches,omitempty"`
+}
+
+func GetReleaseVersion(defaultManifestPath string, componentName string) ComponentStatus {
```

Comment:
why not move this function into pkg/cluster/cluster_config.go

Reply:
@zdtsw Earlier i had it in components.go since it is an utility share by component. later on suggestions i moved it to controllers/status/status.go and it looks bit more appropriate, as this is used my components to display releases in status. wdyt?

Reply:
no strong opinion to keep it here or not.